src/Arcus.Scripting.Management/Scripts/Restore-AzApiManagementSoftDeletedService.ps1

```diff
@@ -0,0 +1,46 @@
+Param(
+    [Parameter(Mandatory = $true)][string] $Name = $(throw "Name of the API Management instance is required"),
+    [Parameter(Mandatory = $false)][string] $SubscriptionId = "",
+    [Parameter(Mandatory = $false)][string] $AccessToken = ""
+)
+
+if($SubscriptionId -eq "" -or $AccessToken -eq ""){
+    # Request accessToken in case the script contains records
+    $token = Get-AzCachedAccessToken
+
+    $AccessToken = $token.AccessToken
+    $SubscriptionId = $token.SubscriptionId
+}
+
+$authHeader = @{
+   'Authorization'='Bearer ' + $AccessToken
+}
+
+Write-Host "Checking if the API Management instance with name '$Name' is listed as a soft deleted service"
+$getUri = 'https://management.azure.com/subscriptions/{0}/providers/Microsoft.ApiManagement/deletedservices?api-version=2021-08-01' -f $SubscriptionId
+$deletedServices = (Invoke-RestMethod -Method GET -Uri $getUri -Headers $authHeader)
```

Comment:
Is it an idea to move this to a separate script as it's the same for both scripts? Or would that make things too complicated? Again, not critical for me, though, can be seen as an improvement later.

Reply:
I moved the getting of the soft deleted instances to a seperate script, see latest commit